### PR TITLE
Restrict permissions on designer-assessed metrics

### PR DESCRIPTION
**Pull request by ethn**

### CODE REVIEW

File: mod/metrics/set/abstract/expanded_researched_details.rb [hunk 1]
```diff
@@ -12,45 +12,48 @@ def credit_details
...
-        wrap_with(:div, _render_sources, class: "cited-sources mt-3"),
-        overridden_calculated_value
-      ]
+      [checked_by_details, source_details, override_details]
     end
   end
 
-  def overridden_calculated_value
+  def checked_by_details
+    return if metric_card.designer_assessed?
```
Comment (ethn) [inline mod/metrics/set/abstract/expanded_researched_details.rb:21]: this reorganization was about making checked by details _not_ show up on designer assessed metrics.

File: mod/isic/spec/set/self/isic_section_formula_spec.rb [hunk 2]
```diff
@@ -10,7 +10,9 @@
 
   describe "integration" do
     it "makes all the calculations between class and section" do
-      create_answer metric: Card["OpenCorporates+Industry Class"], content: %w[0111 9609]
+      create_answer metric: Card["OpenCorporates+Industry Class"],
+                    content: %w[0111 9609],
+                    user: "Joe Admin"
```
Comment (ethn) [inline mod/isic/spec/set/self/isic_section_formula_spec.rb:15]: (This is designer assessed, so after the changes, Joe User couldn't create answers for it any more)

File: mod/metrics/set/type/metric.rb [hunk 3]
```diff
@@ -62,5 +68,5 @@ def answer_card company, year
 end
 
 def designer_assessed?
-  research_policy.casecmp("designer assessed").zero?
+  research_policy.tr("[]", "").casecmp("designer assessed").zero?
```
Comment (ethn) [inline mod/metrics/set/type/metric.rb:71]: wasn't working before.

File: mod/researched_metrics/spec/set/abstract/designer_permissions_spec.rb [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    specify "designer can #{action} metric" do
+      designer_can action, metric
+    end
+
+    specify "nondesigner cannot #{action} metric" do
+      nondesigner_cant action, metric
+    end
+
+    specify "super user can #{action} metric" do
+      Card::Auth.as "Joe Admin" do
```
Comment (xithan) [inline mod/researched_metrics/spec/set/abstract/designer_permissions_spec.rb]: We have a shorthand for this. Search for `with_user:` to see an example

File: mod/researched_metrics/spec/set/abstract/designer_permissions_spec.rb [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+      specify "designer can update answer field: #{field}" do
+        designer_can :update, answer_field_card(field)
+      end
+
+      specify "nondesigner can't update answer field: #{field}" do
+        nondesigner_cant :update, answer_field_card(field)
+      end
+    end
+  end
+end
```
Comment (xithan) [inline mod/researched_metrics/spec/set/abstract/designer_permissions_spec.rb:83]: nice tests 👍

File: spec/spec_helper.rb [hunk 6]
```diff
@@ -16,10 +16,10 @@
 include SourceHelper
 include SharedData::Samples
 
-def create_answer metric: sample_metric, company: sample_company,
+def create_answer metric: sample_metric, company: sample_company, user: "Joe User",
                   content: "content", year: "2015", source: sample_source.name
   #content ||= "I'm fine, I'm just not happy."
```
Comment (xithan) [inline spec/spec_helper.rb]: I think this emotional state is from Henry and probably outdated and can be deleted :wink:

**Review by xithan — APPROVED** [review]

**Pull request merged**